src/Core/Managed/Shared/Extensibility/TelemetrySink.cs

```diff
@@ -0,0 +1,149 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility
+{
+    using System;
+    using System.Diagnostics;
+    using System.Threading;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+    /// <summary>
+    /// Represents a destination for telemetry, consisting of a set of telemetry processors and a channel.
+    /// </summary>
+    public sealed class TelemetrySink : IDisposable, ITelemetryProcessor
+    {
+        private TelemetryConfiguration telemetryConfiguration;
+        private ITelemetryChannel telemetryChannel;
+        private bool shouldDisposeChannel;
+        private TelemetryProcessorChain telemetryProcessorChain;
+        private TelemetryProcessorChainBuilder telemetryProcessorChainBuilder;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TelemetrySink"/> class.
+        /// </summary>
+        /// <param name="telemetryConfiguration">Telemetry configuration to use for the new <see cref="TelemetrySink"/> instance.</param>
+        /// <param name="telemetryChannel">Telemetry channel to use for the new <see cref="TelemetrySink"/> instance.</param>
+        public TelemetrySink(TelemetryConfiguration telemetryConfiguration, ITelemetryChannel telemetryChannel = null)
```

Comment:
OK will make sure TelemetrySink is not an ITelemetryProcessor.

It is not a ITelemetryChannel either (since it can have its own set of telemetry processors), so I am thinking it will have its own `Process` method.